3-limit-service-time/main.go

```diff
@@ -21,10 +25,19 @@ type User struct {
 // HandleRequest runs the processes requested by users. Returns false
 // if process had to be killed
 func HandleRequest(process func(), u *User) bool {
+	if u.TimeUsed >= 10 {
+		return false
+	}
+	
+	timestamp := time.Now().Unix()
+	
 	process()
-	return true
+	
+	u.TimeUsed = time.Now().Unix() - timestamp 
```

Comment:
pls work with time.Time/time.Duration instead of int64 timestamp directly 
it's just safer in general + you've got added bonus of handling timezones and DST correctly